Fix an out of bounds access in ForwardMolSupplier

bp-kelley opened the pull request:
An update to the osx toolchain on conga hit this long standing bug,

greglandrum commented on Code/GraphMol/FileParsers/ForwardSDMolSupplier.cpp:
```diff
@@ -116,8 +116,8 @@ void ForwardSDMolSupplier::readMolProps(ROMol &mol) {
           int nplines = 0;  // number of lines for this property
-          while (stmp.length() != 0 || tempStr[0] == ' ' ||
-                 tempStr[0] == '\t') {
+          while (stmp.length() != 0 || (tempStr.length() && (tempStr[0] == ' ' ||
```
Please use .empty() instead of comparing length() with zero

greglandrum reviewed (changes requested):
Minor changes

greglandrum commented:
@bp-kelley The Mac tests are still (again?) failing

bp-kelley commented:
@greglandrum I fixed the remaining errors.  I also switched from string[0] to string.at(0) to help me find the issues.  We can switch this back if you like.

bp-kelley commented:
@greglandrum yep, it's fixed

greglandrum reviewed (approved):
LGTM
